keps/sig-scheduling/3633-matchlabelselectors-to-podaffinity/README.md

```diff
@@ -369,7 +376,7 @@ type PodAffinityTerm struct {
   // which pods will be taken into consideration for the incoming pod's pod (anti) affinity. 
   // The default value is empty.
   // +optional
-  MatchLabelSelectors []strinMatchLabelSelectorg
+  MatchLabelSelectors []MatchLabelSelector
```

Comment:
Continuing the discussion from:
https://github.com/kubernetes/enhancements/pull/4001#discussion_r1217797463

> Ah, yes correct. This PR changes MatchLabelSelectors to be handled in kube-apiserver when Pod is created.
So, existing objects will continue to have the labelselector made by MatchLabelSelectors even after the feature gate is turned off.
>
> What is the ideal action here? OK to leave the existing objects when turning off the feature gate (as long as we state in KEP), or we somehow need to ignore the labelselector made by MatchLabelSelectors (gonna be tricky though..)?

Yes - I'm fine with leaving the existing objects as is [in fact we don't really have good mechanisms to do anything smarter].

Just please extend the `Downgrade` section with explanation about:
- leaving the existing objects as is
- scheduler ignoring this field when FG is disabled (even if set)
- IIUC, also scheduler depends on the FG, so please update the first question in PRR

Once all of the above is applied, this seems good to go for Alpha (IIUC we're only targeting Alpha now)

Reply:
OK, thanks for clarifying.

Reply:
> scheduler ignoring this field when FG is disabled (even if set)
IIUC, also scheduler depends on the FG, so please update the first question in PRR

Regarding them, No. 
The kube-apiserver generates `LabelSelector` from `MatchLabelSelectors` and kube-scheduler computes PodAffinity/PodAntiAffinity as it does now. So, kube-scheduler doesn't look at `MatchLabelSelectors` directly and it doesn't depend on the feature gate. (instead, it just looks at `LabelSelector` generated from kube-apiserver.)